File: StudentCredit.Application/Reports/Dtos/Reports/SearchReportFilter.cs
using StudentCredit.Application.Reports.Enums;
using StudentCredit.Data.Applications.Common.Enums;
using StudentCredit.Data.Nomenclatures.Enums;

namespace StudentCredit.Application.Reports.Dtos.Reports
{
    public class SearchReportFilter
    {
        public ReportType ReportType { get; set; }
        //When ReportType == ReportType.NewContract
        public NewContractFilterType? NewContractFilterType { get; set; }
        public LearnerType? LearnerType { get; set; }
        public InstitutionOwnershipType? InstitutionOwnershipType { get; set; }

        public int? BankId { get; set; }
        public CreditType? CreditType { get; set; }
        public int? ContractYear { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public int Limit { get; set; } = 10;
        public int Offset { get; set; } = 0;
    }
}

File: WebApplication/src/modules/report/models/search-report.filter.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { BaseSearchFilter } from "src/infrastructure/services/base-search.filter";
import { CreditType } from "src/modules/application/common/enums/credit-type.enum";
import { ReportType } from "../enums/report-type.enum";
import { NewContractFilterType } from "../enums/new-contract-filter-type.enum";
import { LearnerType } from "../enums/learner-type.enum";
import { InstitutionOwnershipType } from "src/modules/nomenclature/enums/institution-ownership-type.enum";

export class SearchReportFilter extends BaseSearchFilter {
  reportType: ReportType;
  newContractFilterType: NewContractFilterType;
  learnerType: LearnerType;
  institutionOwnershipType: InstitutionOwnershipType;
  bank: NomenclatureDto;
  bankId: number | null;
  contractYear: number | null;
  creditType: CreditType | null = null;

  startDate: Date;
  endDate: Date;

  constructor() {
    super(100);
  }
}
